libs/orchestrator-ui-components/src/lib/components/Table/TableSettingsModal/TableSettingsModal.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState } from 'react';
+import {
+    EuiForm,
+    EuiFormRow,
+    EuiHorizontalRule,
+    EuiSelect,
+    EuiSpacer,
+    EuiSwitch,
+} from '@elastic/eui';
+import { SettingsModal } from '../../SettingsModal';
+
+export type ColumnConfig<T> = {
+    field: keyof T;
+    name: string;
+    isVisible: boolean;
+};
+
+export type TableConfig<T> = {
+    columns: ColumnConfig<T>[];
+    selectedPageSize: number;
+};
+
+export type TableSettingsModalProps<T> = {
+    tableConfig: TableConfig<T>;
+    pageSizeOptions: number[];
+    onClose: () => void;
+    onUpdateTableConfig: (updatedTableConfig: TableConfig<T>) => void;
+    onResetToDefaults: () => void;
+};
+
+export const TableSettingsModal = <T,>({
```

Comment:
I know this is not what you find in most docs but I'm a fan of a bit more descriptive general types so TABLE_COLUMNS instead of T

Reply:
Let's discuss this during the next team meeting. When we make a decision we can apply the same style everywhere in the project